### ISSUE TITLE
tf.keras.activations.swish has wrong gradient

### ISSUE THREAD

**Issue by VictoriaGriffith** [keras-team/keras#16962]
This issue was originally reported in https://github.com/tensorflow/tensorflow/issues/57481

 <details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

tf 2.9

### Custom Code

Yes

### OS Platform and Distribution

Linux Ubuntu 20.04

### Python version

3.9

### Current Behaviour?

In the example, `tf.keras.activations.swish` accepts a complex input, and `tf.test.compute_gradient` checks the theoretical and numerical gradient, and the two gradients are different. For example, on [0,1], the theoretical gradient is `0.56998741`, and the numerical gradient is `-0.56998753`.

Probably related to https://github.com/tensorflow/tensorflow/issues/57357

### Standalone code to reproduce the issue

```shell
import tensorflow as tf
x = tf.complex(tf.random.uniform([1, 20, 1], maxval=0, dtype=tf.float64),tf.random.uniform([1, 20, 1], maxval=1.000000, dtype=tf.float64))

def f(x):
    return tf.keras.activations.swish(x, )
theoretical, numerical = tf.test.compute_gradient(f, [x])
print(theoretical)
print(numerical)

assert tf.experimental.numpy.allclose(theoretical, numerical, atol=1e-3, rtol=1e-3)
```

### Relevant log output

```shell
(array([[ 0.5       ,  0.56998741,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [-0.56998741,  0.5       ,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.5       , ...,  0.        ,
         0.        ,  0.        ],
       ...,
       [ 0.        ,  0.        ,  0.        , ...,  0.5       ,
         0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.5       ,  0.1744664 ],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
        -0.1744664 ,  0.5       ]]),)
(array([[ 0.5       , -0.56998753,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [ 0.56998728,  0.5       ,  0.        , ...,  0.        ,
         0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.5       , ...,  0.        ,
         0.        ,  0.        ],
       ...,
       [ 0.        ,  0.        ,  0.        , ...,  0.5       ,
         0.        ,  0.        ],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.5       , -0.17446643],
       [ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.17446637,  0.5       ]]),)
AssertionError
```
</details></details>

**Comment by tilakrayal** [comment]
@gadagashwini,
I was able to reproduce the issue on tensorflow v2.8, v3.9 and nightly. Kindly find the gist of it [here](https://colab.research.google.com/gist/tilakrayal/d6ee26fcd39238c4b023a9fcb154128c/untitled559.ipynb).

**Comment by cantonios** [comment]
Numerical precision issue - there are singularities, so the numerical gradient loses accuracy.

**Comment by rchao** [comment]
Since the original issue on tensorflow github has been closed, closing this as well. Please reopen if the issue is still there.
